core/include/kernel/notif.h

```diff
@@ -106,6 +106,11 @@ void notif_free_async_value(uint32_t value);
  */
 TEE_Result notif_wait(uint32_t value);
 
+/*
+ * Wait timeout in normal world for a value to be sent by notif_send()
+ */
+TEE_Result notif_wait_timeout(uint32_t value, uint32_t tmo);
```

Comment:
It may be better to split in two patches: extending condvars then modifying notifications.

Reply:
ok, I will spilt it.